services/src/main/scala/cromwell/pipeline/service/GitLabProjectVersioning.scala

```diff
@@ -46,7 +47,27 @@ class GitLabProjectVersioning(httpClient: HttpClient, config: GitLabConfig)
   override def getFileTree(project: Project, version: Option[Version])(
     implicit ec: ExecutionContext
   ): AsyncResult[List[String]] = ???
+
   override def getFile(project: Project, path: Path, version: Option[Version])(
     implicit ec: ExecutionContext
-  ): AsyncResult[String] = ???
+  ): AsyncResult[String] = {
+    val filePath: String = URLEncoder.encode(path.toString, "UTF-8")
+    val fileVersion: String = version.map(el => el.value).getOrElse("master")
+
+    val responseFuture = httpClient.get(
+      s"${config.url}/projects/${project.repository}/repository/files/$filePath/raw",
+      Map("ref" -> fileVersion),
+      config.token
+    )
+
+    responseFuture
+      .map(resp => {
+        if (resp.status == StatusCodes.OK.intValue) {
+          Right(resp.body)
```

Comment:
Somehow my comment to for this line wasn't added to review.

We have `ProjectFile` entity. And I guess that we need to return it instead of string.

Reply:
Done